src/EbmlUnicodeString.cpp

```diff
@@ -44,49 +39,34 @@ UTFstring & UTFstring::operator=(const UTFstring & _aBuf)
   return *this;
 }
 
-UTFstring::operator const wchar_t*() const {return _Data;}
-
-
 UTFstring & UTFstring::operator=(const wchar_t * _aBuf)
 {
-  delete [] _Data;
-  if (_aBuf == nullptr) {
-    _Data = new wchar_t[1];
-    _Data[0] = 0;
-    UpdateFromUCS2();
-    return *this;
-  }
+  if (_aBuf != nullptr)
+    WString = _aBuf;
+  else
+    WString.clear();
 
-  std::size_t aLen;
-  for (aLen=0; _aBuf[aLen] != 0; aLen++);
-  _Length = aLen;
-  _Data = new wchar_t[_Length+1];
-  for (aLen=0; _aBuf[aLen] != 0; aLen++) {
-    _Data[aLen] = _aBuf[aLen];
-  }
-  _Data[aLen] = 0;
   UpdateFromUCS2();
   return *this;
 }
 
 UTFstring & UTFstring::operator=(wchar_t _aChar)
 {
-  delete [] _Data;
-  _Data = new wchar_t[2];
-  _Length = 1;
-  _Data[0] = _aChar;
-  _Data[1] = 0;
+  WString = _aChar;
   UpdateFromUCS2();
   return *this;
 }
 
 bool UTFstring::operator==(const UTFstring& _aStr) const
 {
-  if ((_Data == nullptr) && (_aStr._Data == nullptr))
-    return true;
-  if ((_Data == nullptr) || (_aStr._Data == nullptr))
+  // Only compare up to the first 0 char in both strings.
+  auto LengthThis  = std::distance(WString.begin(),       std::find(WString.begin(),       WString.end(),       L'\0'));
```

Comment:
These can use WString.find() directly.

Reply:
I know, just found the oneliners to be optically pleasing. Anyway, I've pushed a commit that moves length-to-first-null calculation into its own function, using `std::wstring::find`.